src/models/user.js

```diff
@@ -1,9 +1,136 @@
-import mongoose from 'mongoose';
+import { mongoose } from 'mongoose';
+import { bcrypt } from 'bcrypt-nodejs';
+import { validate } from 'mongoose-validator';
+import { isValidPassword } from 'mongoose-custom-validators';
 
-const UserSchema = new mongoose.Schema({
-  name: 'string',
-  type: 'string',
+const { SALT_INDEX } = 10;
+const { Schema } = mongoose.Schema;
+
+const nameValidator = [
+  validate({
+    validator: 'isLength',
+    arguments: [2, 32],
+    message: 'Name should be 2 to 32 characters',
+  }),
+  validate({
+    validator: 'isAlphabetical',
+    arguments: ['/^[a-zA-Z]*$/+[^s-]'],
+    message: 'Name should contain characters from A - Z',
+  }),
+
+  validate({
+    validator: 'isNumeric',
+    no_symbols: false,
+    message: 'There can be no +, -, or .',
+  }),
+];
+
+const emailValidator = [
+  validate({
+    type: String,
+    valiator: 'isLength',
+    arguments: [3, 32],
+    message: 'Name should be btween {ARGS[0]} and {ARGS[1]} characters',
+  }),
+
+  validate({
+    validator: 'isEmail',
+    message: 'Email address invalid',
+  }),
+];
+
+const passwordValidator = [
+  validate({
+    validator: 'isLength',
+    arguments: [8, 32],
+    message: 'Password needs to be between {ARGS[0]} to {ARGS[1]} long',
+  }),
+
+  validate({
+    valiator: str => isValidPassword(str, { minlength: 10 }),
+    message:
+      'Password must have at least: 1 uppercase letter, 1 lowercase letter, 1 number, and 1 special character.',
+  }),
+];
+
+const UserSchema = new Schema({
+  name: {
```

Comment:
Thoughts on just making `first_name` and `last_name` top-level fields and not nested under `name`? seems unnecessary to nest them. We can add `name` as a virtual property to the schema instead.

We should also use camelCase for field names 😄